src/java/org/apache/cassandra/io/sstable/SSTableRewriter.java

```diff
@@ -239,7 +240,10 @@ private void moveStarts(SSTableReader newReader, DecoratedKey lowerbound)
             if (lowerbound.compareTo(latest.last) >= 0)
             {
                 if (!transaction.isObsolete(latest))
+                {
+                    latest.runOnClose(() -> CacheService.instance.obsoleteSSTable(latest.descriptor));
```

Comment:
As this is in `moveStarts`, there are going to be multiple (1000s for large files) copies of the same sstable. The obsoletion list can grow very large. It's preferable to do this on the final deletion or at least use a set.

Best, remove the signal and filter based on the live sstables set.